src/App.tsx

```diff
@@ -15,47 +16,118 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+const getGoodsFromServer = (
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) => {
+  const visibleGoods = [...goods];
+
+  if (sortType === SortType.ALPHABET) {
+    visibleGoods.sort((a, b) => a.localeCompare(b));
+  }
+
+  if (sortType === SortType.LENGTH) {
+    visibleGoods.sort((a, b) => a.length - b.length);
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+};
+
 export const App: React.FC = () => {
+  const [isReversed, setReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+
+  const isAlphabetSort = sortType === SortType.ALPHABET;
+  const isLengthSort = sortType === SortType.LENGTH;
+  const isResetButtonActive = sortType !== SortType.NONE || isReversed;
+
+  const products = getGoodsFromServer(
+    goodsFromServer,
+    { sortType, isReversed },
+  );
+
+  const sortByAlphabet = () => {
+    setSortType(SortType.ALPHABET);
+  };
+
+  const sortByLength = () => {
+    setSortType(SortType.LENGTH);
+  };
+
+  const reverse = () => {
+    setReversed(current => !current);
+  };
+
+  const reset = () => {
+    setReversed(false);
+    setSortType(SortType.NONE);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={classNames('button is-info', {
+            'is-light': !isAlphabetSort,
+          })}
+          onClick={sortByAlphabet}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={classNames('button is-success', {
+            'is-light': !isLengthSort,
+          })}
+          onClick={sortByLength}
```

Comment:
```suggestion
          onClick={handleSortByLength}
```
![image](https://user-images.githubusercontent.com/87998640/233131000-d022198a-c797-423a-8b92-2cdac44f81e8.png)

Reply:
Thanks for the suggestion but I'd like to hear a mentor's opinion about function names.